corehq/apps/hqwebapp/static/hqwebapp/js/tab_url_hashes.js

```diff
@@ -0,0 +1,36 @@
+/*
+ * Adds URL hash behavior to bootstrap tabs. This enables bookmarking/refreshing and browser back/forward.
+ * Lightly modified from https://stackoverflow.com/questions/18999501/bootstrap-3-keep-selected-tab-on-page-refresh
+ */
+hqDefine("hqwebapp/js/tab_url_hashes", [
+    "jquery",
+], function(
+    $
+) {
+    $(function(){
+        var tabSelector = "a[data-toggle='tab']:not(.no-hash)";
+        if (window.location.hash) {
+            $("a[href='" + window.location.hash + "']").tab('show');
```

Comment:
Does this result in a flash of tab switching on page load? Like it initially loads the first tab and then once the JS loads it flips over to the one in the URL? If yes...

I think it's probably overall a net gain to be able to link to tabs, so I'm not arguing to remove this. However this can result in usability issues. For example, if someone

- loads the page
- JS is slow to load, so the first (default) tab is selected, not the one in the URL
- click some other tab than the one linked in the URL
- JS loads, switching to the tab linked in the URL
- confusion because the clicked tab is not active

Not sure if there is anything to do about that, but just thought I'd mention it. If nothing else, keep it in the back of your mind in case you ever come up with a better solution that wouldn't have those issues.

Reply:
Hmm, this is a good point. The page will initially load whatever tab, if any, is marked `.active` in the html, then it'll switch when js loads. The accounting app avoids this by not marking any tabs as active. I'll take a pass through app manager to do the same.